### PR TITLE
Refs #24281 - Refactor class methods of create and create! 

### PR DESCRIPTION
**Pull request by ik5**

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -198,28 +198,29 @@ def parse_string_value(val)
...
+    opts ||= { name: new.name }
+    opts.symbolize_keys!
 
-    if (s = Setting.find_by_name(opts[:name].to_s)).nil?
-      column_check(opts)
-      super opts.merge(:value => readonly_value(opts[:name].to_sym) || opts[:value])
-    else
-      create_existing(s, opts)
+    s = Setting.find_by_name(opts[:name].to_s)
+    if s
```
Comment (dLobatog) [inline app/models/setting.rb]: nitpick, `return create_existing(s, opts) if s`

File: app/models/setting.rb [hunk 2]
```diff
@@ -198,28 +198,29 @@ def parse_string_value(val)
     true
   end
 
-  def self.create(opts)
+  # in order to avoid code duplication, this method was introduced
+  def self.create_find_by_name(opts, &block)
```
Comment (dLobatog) [inline app/models/setting.rb]: You may remove the `&block`, implicit is way faster - you may get some background here https://github.com/ruby/ruby/pull/1240

**Review by dLobatog — CHANGES_REQUESTED** [review]
Sorry for the nitpicky comments @ik5 , in general :+1: if tests pass, could you update it? Thank you!

File: app/models/setting.rb (hunk 1 above)
Comment (ik5) [reply app/models/setting.rb]: ack

File: app/models/setting.rb (hunk 2 above)
Comment (ik5) [reply app/models/setting.rb]: keep on forgetting that :( thanks

**Comment by ik5** [comment]
@dLobatog what do you think of this now?

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ik5 !

**Comment by ares** [comment]
@ik5 please don't use Refs in git commit message, it only links the PR in redmine but does not close the issue.

**Pull request merged**